x-pack/plugins/canvas/public/state/actions/elements.js

```diff
@@ -263,7 +263,17 @@ function setExpressionFn({ dispatch, getState }, expression, elementId, pageId,
 
   // read updated element from state and fetch renderable
   const updatedElement = getNodeById(getState(), elementId, pageId);
-  if (doRender === true) {
+
+  // reset element.filter if element is no longer a filter
+  if (
+    updatedElement.filter &&
+    !['dropdownControl', 'timefilterControl', 'exactly'].some(filter =>
```

Comment:
Is there a way we can get the list of filter elements without hard-coding them here? Maybe we can pull the functions from the registry somehow?